packages/headless/src/features/search-parameters/search-parameter-serializer.ts

```diff
@@ -35,7 +38,7 @@ function serializePair(pair: [string, unknown]) {
     return isRangeFacetObject(val) ? serializeRangeFacets(key, val) : '';
   }
 
-  return `${key}${equal}${val}`;
+  return `${key}${equal}${encodeURIComponent(val)}`;
```

Comment:
Atm, the delimiter between params is `&` so using `encodeURIComponent` works well, but I think it comes at a cost that you highlighted in the PR description.

A more fine-grained approach would be to escape just the delimiting character. This would keep uri knowledge outside of the module. We also talked about letting people configure delimiters. Escaping would keep the door open. The logic would work if someone used a character that `encodeURIComponent` does not encode.

If the query is set to `M&G`, and sort criteria is relevance, then:
```
//before
q=M&G&sortCriteria=relevance // splitting on & would give an incorrect result

// after
q=M\&G&sortCriteria=relevance // splitting on unescaped & characters, followed by unescaping delimiters in the value would give a correct result
```

I'm thinking about the future and I don't know if these requirements will come. I think staying flexible is a plus.
If this is not a requirement though, then the current approach should work well,

Reply:
I don't think being able to customize the delimiter is really a need.

What users normally requested to be able to do in JSUI in the past (and were not able to do) is to be able to control the name of the parameters.

eg:

"Instead of `q=foo&sortCriteria=relevance`, I want to have `query=foo&sortedBy=relevance` because I'm also using that in some other system X-Y-Z"